apisix/plugins/error-log-logger.lua

```diff
@@ -0,0 +1,220 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local core = require("apisix.core")
+local nproc = require("ngx.process")
+local errlog = require ("ngx.errlog")
+local batch_processor = require("apisix.utils.batch-processor")
+local plugin_metadata = require("apisix.admin.plugin_metadata")
+local plugin_name = "error-log-logger"
+local table = core.table
+local ngx = ngx
+local tcp = ngx.socket.tcp
+local string = string
+local tostring = tostring
+local buffers
+local loaded_flag
+
+local metadata_schema = {
+    type = "object",
+    properties = {
+        host = {type = "string", default = "127.0.0.1"},
+        port = {type = "integer", minimum = 0, default = 9200},
+        tls = {type = "boolean", default = false},
+        tls_options = {type = "string"},
+        timeout = {type = "integer", minimum = 1, default = 3},
+        keepalive = {type = "integer", minimum = 1, default = 30},
+        name = {type = "string", default = plugin_name},
+        level = {type = "string", default = "WARN"},
+        batch_max_size = {type = "integer", minimum = 0, default = 1000},
+        max_retry_count = {type = "integer", minimum = 0, default = 0},
+        retry_delay = {type = "integer", minimum = 0, default = 1},
+        buffer_duration = {type = "integer", minimum = 1, default = 60},
+        inactive_timeout = {type = "integer", minimum = 1, default = 3},
+    },
+    additionalProperties = false,
+}
+
+
+local log_level = {
+    STDERR =    ngx.STDERR,
+    EMERG  =    ngx.EMERG,
+    ALERT  =    ngx.ALERT,
+    CRIT   =    ngx.CRIT,
+    ERR    =    ngx.ERR,
+    ERROR  =    ngx.ERR,
+    WARN   =    ngx.WARN,
+    NOTICE =     ngx.NOTICE,
+    INFO   =    ngx.INFO,
+    DEBUG  =    ngx.DEBUG
+}
+
+
+local config = {
+}
```

Comment:
I think this `local _M = {}` format will look better, what do you think?

Reply:
fixed it